server/src/main/java/com/netflix/conductor/bootstrap/BootstrapUtil.java

```diff
@@ -0,0 +1,73 @@
+package com.netflix.conductor.bootstrap;
+
+import com.netflix.conductor.dao.IndexDAO;
+import com.netflix.conductor.elasticsearch.EmbeddedElasticSearch;
+import com.netflix.conductor.grpc.server.GRPCServer;
+import org.apache.log4j.PropertyConfigurator;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+
+public class BootstrapUtil {
+    public static void startEmbeddedElasticServer(Optional<EmbeddedElasticSearch> embeddedSearchInstance) {
```

Comment:
It would be better to not invoke this method if embedded elasticsearch instance is not found.
This would also clean up not having to pass `Optional` variables as parameters.

Basically, in the `Main` class
```java
Optional<EmbeddedElasticSearch> embeddedElasticsearchInstance = serverInjector.getInstance(EmbeddedElasticSearchProvider.class).get();
        if (embeddedElasticsearchInstance.isPresent()) {
                   BootstrapUtil.startEmbeddedElasticsearchServer(embeddedElasticsearchInstance.get());
}
```

Reply:
@apanicker-nflx  Thanks for your suggestions. Added checks in `Main` and in servlet context initialization to avoid passing `Optional` variables in method parameters.